extendable_fastapi/tests/test_strict_extendable_base_model.py

```diff
@@ -0,0 +1,64 @@
+from datetime import date
+
+from extendable_pydantic import ExtendableBaseModel
+
+from pydantic import ValidationError
+
+from ..schemas import StrictExtendableBaseModel
+from .common import FastAPITransactionCase
+
+
+class TestStrictExtendableBaseModel(FastAPITransactionCase):
+    class Model(ExtendableBaseModel):
+        x: int
+        d: date | None
+
+    class StrictModel(StrictExtendableBaseModel):
+        x: int
+        d: date | None
+
+    def test_Model_revalidate_instance_never(self):
+        # Missing required fields but no re-validation
+        m = self.Model.model_construct()
+        self.assertEqual(m.model_validate(m).model_dump(), {})
+
+    def test_StrictModel_revalidate_instance_always(self):
+        # Missing required fields and always revalidate
+        m = self.StrictModel.model_construct()
+        with self.assertRaises(ValidationError):
+            m.model_validate(m)
+
+    def test_Model_validate_assignment_false(self):
+        # Wrong assignment but no re-validation at assignment
+        m = self.Model(x=1, d=None)
+        m.x = "TEST"
```

Comment:
This triggers a warning in the tests: 

```WARNING odoo py.warnings: /opt/odoo-venv/lib/python3.10/site-packages/pydantic/main.py:301: UserWarning: Pydantic serializer warnings:
  Expected `int` but got `str` - serialized value may not be as expected```
  
  This is coherent with the test case but we maybe want to avoid having this warning every time the tests are running?